ta4j-core/src/main/java/org/ta4j/core/num/DoubleNum.java

```diff
@@ -23,47 +23,73 @@
  */
 package org.ta4j.core.num;
 
-import static org.ta4j.core.num.NaN.NaN;
-
+import java.util.Map;
+import java.util.Objects;
+import java.util.WeakHashMap;
 import java.util.function.Function;
 
+import static org.ta4j.core.num.NaN.NaN;
+
 /**
  * Representation of Double. High performance, lower precision.
  *
  * @apiNote the delegate should never become a NaN value. No self NaN checks
  *          provided
  */
 public class DoubleNum implements Num {
+    private static final DoubleNum DZERO = new DoubleNum(0d);
+    private static final DoubleNum DNZERO = new DoubleNum(-0.0d);
+    private static final DoubleNum D1_0 = new DoubleNum(1d);
+    private static final DoubleNum DHUNDRED = DoubleNum.valueOf(100d);
+
+    public static Map<Double, DoubleNum> cache = new WeakHashMap<>();
+    public static boolean useCache = Objects.equals(System.getenv("TA4J_DOUBLENUM_CACHE"), "true"); // false default
+
+    static DoubleNum bind(double delegate) {
+        if (0d == delegate)
+            return DZERO;
+        if (1d == delegate)
+            return D1_0;
+        if (-0.0d == delegate)
+            return DNZERO;
+        if (!useCache)
+            return new DoubleNum(delegate); // this should be a basic case for the C2 var profiler inlining
```

Comment:
The sentence "this should be a basic case for the C2 var profiler inlining" is cryptic..is this comment really  neccessary?